c++/src/kj/async-io-unix.c++

```diff
@@ -1487,7 +1487,11 @@ kj::Own<PeerIdentity> SocketAddress::getIdentity(kj::LowLevelAsyncIoProvider& ll
       // MacOS / FreeBSD
       struct xucred creds;
       uint length = sizeof(creds);
+#if defined SOL_LOCAL
```

Comment:
Perhaps cleaner to have this at the top of the file instead?

```
#ifndef SOL_LOCAL
// DragonFly or FreeBSD < 12.2
#define SOL_LOCAL 0
#endif
```

Reply:
I agree, and also I would want to restrict this to the operating systems where it is known to be correct, so...

#if !defined(SOL_LOCAL) && (__FreeBSD__ || __DragonflyBSD__)
// On DragonFly or FreeBSD < 12.2 you're supposed to use 0 for SOL_LOCAL.
#define SOL_LOCAL 0
#endif